### PR TITLE
Installation models folder review and cs

### PR DESCRIPTION
**Pull request by zero-24**
#### Summary of Changes

Code Review and CS for the models folder of the installation
#### Testing Instructions

Confirm that the installation still works with this branch: https://github.com/zero-24/joomla-cms/archive/installmodelcs.zip

### CODE REVIEW

File: installation/model/database.php [hunk 1]
```diff
@@ -808,6 +823,10 @@ protected function postInstallSampleData($db)
 	 */
 	public function backupDatabase($db, $prefix)
 	{
+		// Get the application.
+		/* @var InstallationApplicationWeb $app */
+		$app = JFactory::getApplication();
```
Comment (mbabker) [inline installation/model/database.php]: Personally I'd suggest reverting this `$app` change.  Though the app gets referenced a couple of times in the method, it's only in error conditions.  This causes the reference to always be loaded (you're saving a couple CPU cycles and a small bit of memory not loading it needlessly).

File: installation/model/languages.php [hunk 2]
```diff
@@ -489,10 +495,7 @@ public function setDefault($language, $cms_client = 'administrator')
 	 */
 	public function getOptions()
 	{
-		$session = JFactory::getSession();
-		$options = $session->get('setup.options', array());
-
-		return $options;
+		return JFactory::getSession();->get('setup.options', array())
```
Comment (mbabker) [inline installation/model/languages.php]: You've got your semicolon in the wrong spot here.

File: installation/model/setup.php [hunk 3]
```diff
@@ -25,10 +25,7 @@ class InstallationModelSetup extends JModelBase
 	 */
 	public function getOptions()
 	{
-		$session = JFactory::getSession();
-		$options = $session->get('setup.options', array());
-
-		return $options;
+		return JFactory::getSession()->get('setup.options', array());;
```
Comment (mbabker) [inline installation/model/setup.php]: Extra semicolon.

File: installation/model/languages.php (hunk 2 above)
Comment (zero-24) [inline installation/model/languages.php]: thanks fixed

File: installation/model/database.php (hunk 1 above)
Comment (zero-24) [inline installation/model/database.php]: I have removed it from the whole file see: https://github.com/joomla/joomla-cms/pull/10361/commits/3b956f473dac31afde50973b250f113945af206e

File: installation/model/setup.php (hunk 3 above)
Comment (zero-24) [inline installation/model/setup.php]: fixed thanks.

**Comment by zero-24** [comment]
Same here it looks like this is realy hard to test? Please let me know what is unclear than I can add a better desciption ;)

**Comment by bertmert** [comment]
I have tested this item :white_check_mark: successfully on 82106fbafd4f711ea93c575e11362043b7c15292<br /><br />Tested install. 1. install with demo datas and 2. install multilang.
No issues.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10361">issues.joomla.org/joomla-cms/10361</a>.</sub>

**Comment by bertmert** [comment]
I have tested this item :white_check_mark: successfully on 4a96a35632f29f050c02c18be9d498a9bcccfe12<br /><br />Just code review because only a language string has been changed since my last successful test + language string exists in installation/language/en-GB/en-GB.ini.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10361">issues.joomla.org/joomla-cms/10361</a>.</sub>

**Comment by zero-24** [comment]
@bertmert no need to retest just merge conflichts ;)

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 038527ea17e534ce1e51a5899c3fe8f9a84e0106<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10361">issues.joomla.org/joomla-cms/10361</a>.</sub>

**Comment by brianteeman** [comment]
RTC- looks like we have two successful tests and travis is happy - just not updated on the tracker <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10361">issues.joomla.org/joomla-cms/10361</a>.</sub>

**Comment by zero-24** [comment]
Is there any reason to not get the 3.6.3 milestone here?

**Pull request merged**